pulp_service/pulp_service/app/serializers.py

```diff
@@ -44,11 +49,43 @@ class ContentScanSerializer(serializers.Serializer):
     A serializer for package scan.
     """
 
-    repo_version = serializers.CharField()
+    repo_version = serializers.CharField(
+        required=False,
+        allow_blank=True,
+        help_text=_("RepositoryVersion HREF with the packages to be checked."),
+    )
+    package_json = serializers.FileField(
+        required=False,
+        allow_null=True,
+        help_text=_("package-lock.json file with the definition of dependencies to be checked."),
+    )
+
+    def validate(self, data):
+        data = super().validate(data)
+        if bool(data.get("repo_version", None)) == bool(data.get("package_json", None)):
+            raise serializers.ValidationError(
+                _("Exactly one of 'repo_version' or 'package_json' must be specified.")
+            )
+        return data
 
     def validate_repo_version(self, value):
         try:
             repo_version = RepoVersionHrefPrnFilter.get_repository_version(value)
         except:
             raise serializers.ValidationError(_("No matching RepositoryVersion instance found."))
         return repo_version.pk
+
+    def verify_file(self):
+        uploaded_file = self.validated_data["package_json"]
+        temp_file = PulpTemporaryFile.init_and_validate(uploaded_file, None)
```

Comment:
I see you're saving the file as part of validation, but then using that `PulpTemporaryFile` that is already saved to pass to the task via pk. That sounds efficient.

Reply:
+1 to this suggested change. Validate first, then save as `PulpTemporaryFile`